tests/components/sensor/test_recorder.py

```diff
@@ -98,8 +98,6 @@ def set_time_zone():
     # Set our timezone to CST/Regina so we can check calculations
     # This keeps UTC-6 all year round
     dt_util.set_default_time_zone(dt_util.get_time_zone("America/Regina"))
-    yield
-    dt_util.set_default_time_zone(dt_util.get_time_zone("UTC"))
```

Comment:
Should we require the `hass` fixture in this fixture to make sure we restore the timezone to the expected timezone?

Reply:
We can't use the `hass` fixture because some of the tests, which have not yet been converted to coroutines, use the `hass_recorder` fixture

Reply:
This situation seems fragile both before and after this change. I'm not sure we're making it better.